crates/proto/src/serialize/txt/rdata_parsers/aaaa.rs

```diff
@@ -22,14 +22,14 @@ use core::str::FromStr;
 
 use crate::{
     rr::rdata::AAAA,
-    serialize::txt::errors::{ParseError, ParseErrorKind, ParseResult},
+    serialize::txt::errors::{ParseError, ParseResult},
 };
 
 /// Parse the RData from a set of Tokens
 pub(crate) fn parse<'i, I: Iterator<Item = &'i str>>(mut tokens: I) -> ParseResult<AAAA> {
     let address: Ipv6Addr = tokens
         .next()
-        .ok_or_else(|| ParseError::from(ParseErrorKind::MissingToken("ipv6 address".to_string())))
+        .ok_or_else(|| ParseError::MissingToken("ipv6 address".to_string()))
```

Comment:
nit: you've been converting some `ok_or_else`'s to `ok_or` along the way in this diff but there are a few files like this one, `naptr.rs`, and others being touched that leave the `ok_or_else`. Worth a consistency pass?

Reply:
I think the changes made here are guided by lints: creating enum variants directly doesn't count as a non-trivial call that needs to be deferred until necessary, whereas this instance calls `to_string()` which we would prefer not to do unless actually necessary.